apps/aecore/lib/aecore/channel/worker.ex

```diff
@@ -0,0 +1,551 @@
+defmodule Aecore.Channel.Worker do
+  @moduledoc """
+  Module for managing Channels
+  """
+
+  alias Aecore.Channel.ChannelStateOffChain
+  alias Aecore.Channel.ChannelStateOnChain
+  alias Aecore.Channel.ChannelStatePeer
+
+  alias Aecore.Channel.Tx.{
+    ChannelCloseMutalTx,
+    ChannelCloseSoloTx,
+    ChannelSlashTx,
+    ChannelSettleTx,
+    ChannelCreateTx
+  }
+
+  alias Aecore.Tx.{DataTx, SignedTx}
+  alias Aecore.Tx.Pool.Worker, as: Pool
+  alias Aecore.Keys.Wallet
+  alias Aeutil.Events
+
+  use GenServer
+
+  require Logger
+
+  @type role :: :initiator | :responder
+
+  @typedoc """
+  State is map channel_id -> channel_peer_state
+  """
+  @type state :: %{binary() => ChannelStatePeer.t()}
+
+  @type error :: {:error, binary()}
+
+  def start_link(_args) do
+    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
+  end
+
+  def init(_args) do
+    Events.subscribe(:new_top_block)
+    {:ok, %{}}
+  end
+
+  @doc """
+  Notifies channel manager about new mined tx
+  """
+  def new_tx_mined(%SignedTx{data: %DataTx{type: ChannelCreateTx}} = tx) do
+    opened(tx)
+  end
+
+  def new_tx_mined(%SignedTx{data: %DataTx{type: ChannelCloseMutalTx}} = tx) do
+    closed(tx)
+  end
+
+  def new_tx_mined(%SignedTx{data: %DataTx{type: ChannelCloseSoloTx}} = _tx) do
+    # TODO
+    :ok
+  end
+
+  def new_tx_mined(%SignedTx{data: %DataTx{type: ChannelSlashTx}} = _tx) do
+    # TODO
+    :ok
+  end
+
+  def new_tx_mined(%SignedTx{data: %DataTx{type: ChannelSettleTx}} = tx) do
+    closed(tx)
+  end
+
+  def new_tx_mined(%SignedTx{}) do
+    # We don't care about this tx
```

Comment:
Should it be like this or we should expect some functionality here?

Reply:
Please read PR description. This is in TODO. There is no clear way how to implement this right now.

Reply:
Yes it should be like this. We don't do anything with other types of txs.